aui.views/src/AUI/Action/MenuProvider/AWindowMenuProvider.cpp

```diff
@@ -76,17 +90,29 @@ class AMenuContainer: public AViewContainer {
 
                     auto items = i.subItems;
                     if (i.enabled) {
-                        connect(view->mouseEnter, [&, view, items] {
+                        connect(view->mouseEnter, [this, view, items] {
+                            mTimer->fired.clearAllConnections();
+                            mTimer->stop();
+
                             if (mSubWindow) {
                                 mSubWindow->close();
                             }
 
-                            mSubWindow = _new<AMenuContainer>(items);
-                            auto pos = getPositionInWindow();
-                            auto window = getWindow();
-                            auto surfaceContainer = window->createOverlappingSurface(pos, mSubWindow->getMinimumSize());
-                            surfaceContainer->setLayout(_new<AStackedLayout>());
-                            surfaceContainer->addView(mSubWindow);
+                            connect(mTimer->fired, [this, view, items] {
+                                auto pos = mOriginPosition + view->getPosition() + glm::ivec2(getMinimumSize().x, 0);
+                                mSubWindow = _new<AMenuContainer>(items, pos);
+                                ABaseWindow* window;
+                                if (auto s = mSurface.lock()) window = s->getParentWindow();
+                                else window = AWindow::current();
+                                auto surfaceContainer = window->createOverlappingSurface(pos, mSubWindow->getMinimumSize());
+                                surfaceContainer->setLayout(_new<AStackedLayout>());
+                                surfaceContainer->addView(mSubWindow);
+                                mSubWindow->setSurface(surfaceContainer);
+
+                                mTimer->stop();
```

Comment:
Should I move the `mouseEnter` callback or the `fired` callback? Also since both functions captures `view` and `items` they will be passed as argument of the new member function right ? Will look to something like:
```cpp
connect(mSublistOnHoverDisplayDelay->fired, [this, view, items] {
    onShowSublistMenu(view, items);
});
```

Reply:
Yeah, this is exactly what I wanted to see